lib/ansible/modules/git.py

```diff
@@ -438,6 +447,17 @@ def get_version(module, git_path, dest, ref="HEAD"):
     return sha
 
 
+def ssh_supports_acceptnewhostkey(module):
+    ssh_path = module.get_bin_path('ssh', True)
```

Comment:
This would introduce a new potential failure point in this module. It probably makes sense, but just worth considering. Would it be confusing that the `git` module fails with an error message stating that `ssh` could not be found?

Reply:
I would change this to `try` and catch `exception` if ssh has not been found, is this ok?

Reply:
https://github.com/ansible/ansible/blob/eb8b3a8479ec82ad622f86ac46f3e9cc083952b8/lib/ansible/module_utils/common/process.py#L16 
`required` is going to be removed too, then I don't think of any other option.

Reply:
I rewrote that function to more Pythonic and deprecated the toggle. The caller should catch `ValueError` and act accordingly. If the binary isn't required, the exception can be ignored. Otherwise, a `fail_json()` or may `warn()` can be used.

I think it would be fine to fail here, but the generic message given by `AnsibleModule.get_bin_path()` may be confusing. A more specific error here explaining why the lack of `ssh` is causing the `git` module to fail is probably more helpful. That's all I was trying to say.